common/cmake/devices.cmake

```diff
@@ -975,6 +975,11 @@ function(ADD_FPGA_TARGET)
     if(NOT "${ADD_FPGA_TARGET_INPUT_IO_FILE}" STREQUAL "")
       add_file_target(FILE ${ADD_FPGA_TARGET_INPUT_IO_FILE})
     endif()
+    if(NOT "${ADD_FPGA_TARGET_INPUT_XDC_FILE}" STREQUAL "")
+      add_file_target(FILE ${ADD_FPGA_TARGET_INPUT_XDC_FILE})
+      get_file_location(INPUT_XDC_FILE ${ADD_FPGA_TARGET_INPUT_XDC_FILE})
+      get_target_property(PART_JSON ${BOARD} PART_JSON)
```

Comment:
Should be using `get_target_property_required`

`PART_JSON` will not be set if `EXPLICIT_ADD_FILE_TARGET` is set